### PR TITLE
Firefly-1406: fixes 1, 29, 54 (partial, not all tables), 62, 64

### PR DESCRIPTION
**Pull request by robyww**
#### [Firefly-1406](https://jira.ipac.caltech.edu/browse/FIREFLY-1406): fixes 1, 29, 54 (partial, not all tables), 62, 64

 - from: https://confluence.ipac.caltech.edu/display/IRSA/Testing+Feb+2024+-+Joy+UI%2C+mostly
 - fixes three color tab infinate loop
 - add pipes between tabs
 - tab panel fixes

#### Testing
- https://firefly-1406-fixes.irsakudev.ipac.caltech.edu/irsaviewer
- https://firefly-1406-fixes.irsakudev.ipac.caltech.edu/irsaviewer/dce
- https://fireflydev.ipac.caltech.edu/firefly-1406-fixes/firefly
- image borders and some table border are round
- tabs have pipes between 

check when confirmed:

- [x] 1
- [x] 29
- [ ] 54 (partial)
- [x] 62
- [x] not 64, 63
- [x] tabs have pipes between
- [ ] 3 color panel works

### CODE REVIEW

**Comment by lrebull** [comment]
Not sure which parts of 54 i should check. TST seems still differentiated; i do see a box around the chart now in IV and DCE.

**Comment by lrebull** [comment]
video for 64, both in my original  window and 'new private window'

https://github.com/Caltech-IPAC/firefly/assets/23038600/9b6430d8-c02b-43f1-b751-b5a9caccfa6b

**Comment by lrebull** [comment]
wow, i guess i forgot to test the 3-color search before! but it's not working for me at all, at least when i already have search results in irsa viewer. it works when you do a 3-color search as the first search i do.

File: src/firefly/js/ui/tap/TableSelectViewPanel.jsx [hunk 1]
```diff
@@ -353,7 +353,7 @@ export function BasicUI(props) {
                         </SplitContent>
                         <SplitContent>
                             { columnsModel ?
-                                <div style={{height:'100%', display:'flex', flexDirection:'column'}}>
+                                <Stack {...{height:1}}>
```
Comment (loitly) [inline src/firefly/js/ui/tap/TableSelectViewPanel.jsx:356]: Simplified to `height={1}`

File: src/firefly/js/templates/fireflyviewer/TriViewPanel.jsx [hunk 2]
```diff
@@ -68,7 +68,7 @@ export const TriViewPanel= memo(( {showViewsSwitch=true, leftButtons, centerButt
                                            expandedMode={expanded===LO_VIEW.tables}/>);
     }
 
-    const isTriView = (showImages||coverageLeft) && (showXyPlots||coverageRight) && showTables;
+    const isTriView = (showImages||coverageLeft) && (showXyPlots||coverageRight) && showTablesInState;
```
Comment (loitly) [inline src/firefly/js/templates/fireflyviewer/TriViewPanel.jsx:71]: what is `showTablesInState`?

**Comment by lrebull** [comment]
63 works!

File: src/firefly/js/visualize/ui/ColorDialog.jsx [hunk 3]
```diff
@@ -132,7 +132,7 @@ function renderStandardThreeColorView(plot,rFields,gFields,bFields) {
     return (
         <Box sx={{pt:.5}}>
             <FieldGroup groupKey={'colorDialogTabs'} keepState={false}>
-                <FieldGroupTabs initialState= {{ value:'red' }} fieldKey='colorTabs'>
+                <FieldGroupTabs initialState= {{ value:'red' }} tabId='colorTabs' fieldKey='colorTabs'>
```
Comment (loitly) [inline src/firefly/js/visualize/ui/ColorDialog.jsx:135]: We can make `tabId` in TabPanel default to a unique ID if not given.

**Review by loitly — APPROVED** [review]
The changes look good.

File: src/firefly/js/visualize/ui/ColorDialog.jsx (hunk 3 above)
Comment (robyww) [reply src/firefly/js/visualize/ui/ColorDialog.jsx:135]: it creates a unique id, that was the problem. It kept re-rendering and creating more ids

File: src/firefly/js/templates/fireflyviewer/TriViewPanel.jsx (hunk 2 above)
Comment (robyww) [reply src/firefly/js/templates/fireflyviewer/TriViewPanel.jsx:71]: that is how the parameter was passing in. I probably should get a better name.

File: src/firefly/js/visualize/ui/ColorDialog.jsx (hunk 3 above)
Comment (loitly) [reply src/firefly/js/visualize/ui/ColorDialog.jsx:135]: I see.  In this case, maybe `FieldGroupTabs` should internally set tabId to fieldKey.

**Pull request merged**